### ISSUE TITLE
File getting uploaded succesfully but getting error in output Task Runner Window

### ISSUE THREAD

**Issue by s-KaiNet** [s-KaiNet/spsave#14]
_From @AvniBhatt on June 14, 2016 10:42_

File is getting uploaded successfully but getting error in Output Task runner window.
**Error**:

```
[16:06:17] Using gulpfile ~\documents\visual studio 2013\Projects\GulpTask\gulpfile.js
[16:06:17] Starting 'concat-js'...
[16:06:17] Finished 'concat-js' after 8.81 ms
[16:06:17] Starting 'minify-js'...
[16:06:17] Finished 'minify-js' after 5.21 ms
[16:06:17] Starting 'upload-to-sp'...
[16:06:17] Finished 'upload-to-sp' after 3.29 ms
[16:06:20] spsave: GulpDemo.js checked out.
[16:06:20] spsave: GulpDemo.js successfully uploaded to '<>/TestDocLib' and checked in. Checkin type: minor
[16:06:20] spsave: Error occured:
[16:06:20] spsave: Error: Argument error, options.body.

[16:06:20] spsave: Stack trace:

[16:06:20] spsave: RequestError: Error: Argument error, options.body.
    at new RequestError (<PATH>\node_modules\gulp-spsave\node_modules\spsave\node_modules\sp-request\node_modules\request-promise\lib\errors.js:14:15)
    at Request.RP$callback [as _callback] (<PATH>\node_modules\gulp-spsave\node_modules\spsave\node_modules\sp-request\node_modules\request-promise\lib\rp.js:60:25)
    at self.callback (<PATH>\node_modules\gulp-spsave\node_modules\spsave\node_modules\sp-request\node_modules\request\request.js:200:22)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at setContentLength (<PATH>\node_modules\gulp-spsave\node_modules\spsave\node_modules\sp-request\node_modules\request\request.js:452:14)
    at Request.init (<PATH>\node_modules\gulp-spsave\node_modules\spsave\node_modules\sp-request\node_modules\request\request.js:457:5)
    at Request.RP$initInterceptor [as init] (<PATH>\node_modules\gulp-spsave\node_modules\spsave\node_modules\sp-request\node_modules\request-promise\lib\rp.js:137:25)
```

**Code:**

```
gulp.task('upload-to-sp', ['minify-js'], function () {
    gulp.src("./Output/*.js")
      .pipe(spsave({
          username: "<UNAME>",
          password: "<PASSWORD>",
          siteUrl: "<URL>",
          folder: "<FOLDER>",
          checkin: true
      }));
});
```

**Note:** 
This issue happens randomly. And occurs during successive builds. For first build, it uploads successfully. During successive builds, it randomly generates this error and file remains checked-out upon process termination. 

If I delete the file again and re-build, it successfully uploads.

_Copied from original issue: s-KaiNet/gulp-spsave#10_

**Comment by s-KaiNet** [comment]
Can you please post full `gulpfile.js`?  
Also check your folder, may be some special symbols are gets copied? For example from log it seems file uploaded to `'<>/TestDocLib'` - that's strange name for folder.

**Comment by s-KaiNet** [comment]
I was able to reproduce your issue. I will take a look today.

**Comment by s-KaiNet** [comment]
@AvniBhatt  so I found the cause.  
This error is thrown when your file is empty. Check if your `minify-js` task produces something.  
Later in the evening I will add additional checking to skip empty file uploads.

**Comment by AvniBhatt** [comment]
**Uploading full gulp.js:**

``` javascript
/// <vs BeforeBuild='minify-js, upload-to-sp' />
//Load gulp
var gulp = require('gulp');
console.log(process.version);
//Load gulp plugins
var concat = require('gulp-concat'); //bundle
var uglify = require('gulp-uglify'); //minify
var rename = require('gulp-rename'); //renaming to min
var spsave = require('gulp-spsave'); //upload to SharePoint

//Bundle JS files together into a single file
gulp.task('concat-js', function () {
    gulp.src("./Scripts/**/*.js")
    .pipe(concat("GulpDemo.js"))
    .pipe(gulp.dest("./Output"))
});

//Minify the GulpDemo.js file. This task has a dependency on concat-js which means concat-js will always run before minify-js
gulp.task('minify-js', ['concat-js'], function () {
    gulp.src("./Output/GulpDemo.js")
    .pipe(uglify())
    .pipe(rename({ suffix: '.min' }))
    .pipe(gulp.dest("./Output"))
});

//Upload GulpDemo.js and GulpDemo.min.js files to SharePoint. This task has a dependency on minify-js which means minify-js will always run before upload-to-sp

gulp.task('upload-to-sp', ['minify-js'], function () {
    gulp.src("./Output/*.js")
      .pipe(spsave({
          username: "<UNAME>",
          password: "<PASSWORD>",
          siteUrl: "URL",
          folder: "TestDocLib",
          checkin: true
      }));
});

//Watcher task. This will watch for any changes in the scenario1.js and scenario2.js files and then run the upload-to-sp task.
gulp.task('watch-js-upload-to-sp', function () {
    gulp.watch("./Scripts/**/*.js", ['upload-to-sp']);
});
```

I checked `minify-js` task produces the minified file.

`TestDocLib` is the name of document library. Please not file is set to upload on root level and not at folder level in document library.

**Comment by s-KaiNet** [comment]
Are you sure about your `concat-js` task below?

```
gulp.task('concat-js', function () {
gulp.src("./Scripts/*/.js")
.pipe(concat("GulpDemo.js"))
.pipe(gulp.dest("./Output"))
});
```

I believe `gulp.src("./Scripts/*/.js")` doesn't grab any file...

**Comment by AvniBhatt** [comment]
Only one file is added here right now. So it grabs just one file.

**Comment by s-KaiNet** [comment]
To make sure this issue is empty file issue, can you please update your gulpfile to include additional info regarding what is in your gulp pipeline: 
1.Install additional plugins `npm install gulp-debug gulp-filesize --save-dev`
2.In top of your gulpfile add:

``` javascript
var size = require('gulp-filesize');
var debug = require("gulp-debug");
```

3.Update `minify-js` task with debug info: 

``` javascript
gulp.task('minify-js', ['concat-js'], function () {
    return gulp.src("GulpDemo.js")
    .pipe(uglify())
    .pipe(rename({ suffix: '.min' }))
    .pipe(debug())
    .pipe(size())
    .pipe(gulp.dest("./out"))
});
```

4.Run minify and see if you pipeline contains files and also take a note on a length of these files.  
**NOTE** Normally gulp.task returns stream, so you need to add return statement for all your task except `watch`

**Comment by s-KaiNet** [comment]
I've added fix for empty files (`spsave 2.1.2`). Now you should see the warning that the file isn't uploaded because it's empty. Can you confirm that?

**Comment by AvniBhatt** [comment]
Thank you. It works now.
